src/go/k8s/pkg/resources/certmanager/pki_test.go

```diff
@@ -0,0 +1,87 @@
+package certmanager_test
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/stretchr/testify/require"
+	"github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources/certmanager"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSchemaRegistryCerts(t *testing.T) {
+	testcases := []struct {
+		name                   string
+		cluster                v1alpha1.Cluster
+		expectedClientCertName string
+		expectedNodeCertName   string
+	}{{
+		name: "SchemaRegistry: When NodeSecretRef is used",
+		cluster: v1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "redpanda",
+				Name:      "fake-cluster",
+			},
+			Spec: v1alpha1.ClusterSpec{
+				Configuration: v1alpha1.RedpandaConfig{
+					SchemaRegistry: &v1alpha1.SchemaRegistryAPI{
+						TLS: &v1alpha1.SchemaRegistryAPITLS{
+							Enabled: true,
+							NodeSecretRef: &v1.ObjectReference{
+								Namespace: "other-namespace",
+								Name:      "custom-secret-ref",
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedClientCertName: "fake-cluster-schema-registry-client",
+		expectedNodeCertName:   "custom-secret-ref",
+	},
+		{
+			name: "SchemaRegistry: When IssuerRef is used",
+			cluster: v1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "redpanda",
+					Name:      "fake-cluster",
+				},
+				Spec: v1alpha1.ClusterSpec{
+					Configuration: v1alpha1.RedpandaConfig{
+						SchemaRegistry: &v1alpha1.SchemaRegistryAPI{
+							TLS: &v1alpha1.SchemaRegistryAPITLS{
+								Enabled: true,
+								IssuerRef: &cmmeta.ObjectReference{
+									Group: "certmanager",
+									Kind:  "ClusterIssuer",
+									Name:  "cluster-issuer",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedClientCertName: "fake-cluster-schema-registry-client",
+			expectedNodeCertName:   "fake-cluster-schema-registry-node",
+		}}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			pki := certmanager.NewPki(nil, &tt.cluster, "FQDN", "clusterFQDN", nil, logr.Discard())
+			// Client cert
+			clientCert := pki.SchemaRegistryAPIClientCert()
```

Comment:
NIT: Should the pki return any Client cert if `RequireClientAuth` is **not** enabled in any of the test cases?

Reply:
I think it should return `nil`, but, I'd have to refactor this function and the callers along with the other ClientCert functions for the other APIs to keep everything consistent.

I actually tried it but I fear breaking other stuff. My proposal is to do this refactor in a separate PR. We can group this with the suggestions from @alenkacz